onnxruntime/core/providers/nv_tensorrt_rtx/nv_execution_provider.cc

```diff
@@ -1502,7 +1529,27 @@ SubGraphCollection_t NvExecutionProvider::GetSupportedList(SubGraphCollection_t
         // When creating model proto from graph viewer, let ORT use priority-based topological sort based on node index.
         // The reason is, in some cases, for example ResNet50, using default topological sort will end up with generating
         // the model proto that has different node ordering compared to original onnx model.
-        graph_viewer->ToProto(*model_proto->mutable_graph(), true, true, 1 /*priority-based topological sort*/);
+
+        // save user provided external data in memory instead of writing to ModelProto
+        // needed for models > 2GB
+        std::vector<TensorrtUserWeights> userWeights;
+        auto allInitializers = graph_viewer->GetAllInitializedTensors();
+
+        if (use_external_data_initializer_) {
+          for (auto& entry : allInitializers) {
+            auto* tp = entry.second;
+            if (tp->has_raw_data()) {
+              userWeights.emplace_back(tp->name(), tp->raw_data());
+            } else if (utils::HasExternalDataInMemory(*tp)) {
+              std::unique_ptr<ONNX_NAMESPACE::TensorProto> full_init;
+              ORT_THROW_IF_ERROR(utils::GetTensorProtoWithDataIfInMemory(*tp, full_init));
+              userWeights.emplace_back(full_init->name(), full_init->raw_data());
+            }
+          }
+        }
```

Comment:
@yuslepukhin can you help us with a better recommendation to handle memory that has external data in memory ? We are copying all raw data into these `TensorrtUserWeights` since otherwise we were seeing some uninitialized memory reads in the case we do not copy but simply store a data pointer and its size.

Especially for data that is external and in memory I suppose there is a much nicer way to do this !

Reply:
As far as the external data in memory is concerned (pointing to OrtValue) one only needs one call and that is utils::GetTensorProtoWithDatIfInMemory() (please, read the doc for it). If the resulting full_init is nullptr, means the data was not in memory, use the original TensorProto (the data can still be on disk though). If full_init is not nullptr, use that TensorProto with data now inlined. No version checks and other unncessary code.

Reply:
Ok I made use of your added API and rebased this branch. Thanks.